client/lib/gsuite/index.js

```diff
@@ -74,25 +74,27 @@ function getAnnualPrice( cost, currencyCode ) {
 }
 
 /**
- * Retrieves the first domain that is eligible to G Suite either from, and in that order:
+ * Retrieves the first domain that is eligible to G Suite in this order:
  *
- *   - The domain from the site currently selected
- *   - The primary domain of the site
+ *   - The domain from the site currently selected, if eligible
+ *   - The primary domain of the site, if eligible
+ *   - The first non-primary domain eligible found
  *
  * @param {String} selectedDomainName - domain name for the site currently selected by the user
  * @param {Array} domains - list of domain objects
- * @returns {String} - Eligible domain name
+ * @returns {String} - the name of the first eligible domain found
  */
 function getEligibleGSuiteDomain( selectedDomainName, domains ) {
 	if ( selectedDomainName && canDomainAddGSuite( selectedDomainName ) ) {
 		return selectedDomainName;
 	}
 
-	const supportedDomains = getGSuiteSupportedDomains( domains );
-
-	const primaryDomain = find( supportedDomains, 'isPrimary' );
+	// Orders domains with the primary domain in first position, if any
+	const supportedDomains = getGSuiteSupportedDomains( domains ).sort( ( a, b ) => {
```

Comment:
Please double check me, but I think you could use [lodash' `sortBy`](https://lodash.com/docs/4.17.15#sortBy) here to write it more concisely (and maybe in a bit more readable way):

```
const supportedDomains = sortBy( getGSuiteSupportedDomains( domains ), ( domain ) => ! domain.isPrimary );
```

Reply:
Excellent! I tested it, and it works great. I only had to remove the negation:

```javascript
const supportedDomains = sortBy( getGSuiteSupportedDomains( domains ), ( domain ) => domain.isPrimary );
```

Reply:
Hmm, that's weird 🤔 I added that negation based on my testing on the REPL in the lodash' docs:

```
var users = [
  { 'domain': 'domain1',   'isPrimary': false },
  { 'domain': 'domain0',   'isPrimary': false },
  { 'domain': 'domain3',   'isPrimary': true },
  { 'domain': 'domain2',   'isPrimary': false },
  { 'domain': 'domain4',   'isPrimary': true },
];
 
_.sortBy(users, (o) => o.isPrimary );

0: Object {domain: "domain1", isPrimary: false}
1: Object {domain: "domain0", isPrimary: false}
2: Object {domain: "domain2", isPrimary: false}
3: Object {domain: "domain3", isPrimary: true}
4: Object {domain: "domain4", isPrimary: true}

_.sortBy(users, (o) => o.isPrimary );

0: Object {domain: "domain3", isPrimary: true}
1: Object {domain: "domain4", isPrimary: true}
2: Object {domain: "domain1", isPrimary: false}
3: Object {domain: "domain0", isPrimary: false}
4: Object {domain: "domain2", isPrimary: false}

```

But maybe I missed something 🤔 (yes, the domain numbers are out of order on purpose to double check lodash is not sorting by them - which we could add, but don't think it matters at this point, since we're basically guessing which domain is more important if none of the custom ones are primary)

Reply:
No, you didn't miss anything. It's me 😅. I was too confident in the unit tests I wrote, and didn't test the case where `isPrimary` is set to `false` (I just removed this property altogether from the domain object for testing). Thank you for being so meticulous 👍.